utilities/transactions/pessimistic_transaction.cc

```diff
@@ -188,12 +188,19 @@ inline Status WriteCommittedTxn::GetForUpdateImpl(
       return s;
     }
   }
-
-  if (!do_validate && kMaxTxnTimestamp != read_timestamp_) {
+  bool enable_udt_validation =
+      txn_db_impl_->GetTxnDBOptions().enable_udt_validation;
+  if (!enable_udt_validation && kMaxTxnTimestamp != read_timestamp_) {
+    return Status::InvalidArgument(
+        "read_timestamp is set but timestamp validation is disabled for the "
+        "DB");
+  } else if (enable_udt_validation && !do_validate &&
+             kMaxTxnTimestamp != read_timestamp_) {
     return Status::InvalidArgument(
         "If do_validate is false then GetForUpdate with read_timestamp is not "
         "defined.");
-  } else if (do_validate && kMaxTxnTimestamp == read_timestamp_) {
+  } else if (enable_udt_validation && do_validate &&
+             kMaxTxnTimestamp == read_timestamp_) {
     return Status::InvalidArgument("read_timestamp must be set for validation");
   }
```

Comment:
Minor but some repeated checks could be eliminated via nesting:

```
if (!enable_udt_validation) {
   if (kMaxTxnTimestamp != read_timestamp_) {
      // ...
   }
} else {
   if (!do_validate) {
      if (kMaxTxnTimestamp != read_timestamp_) {
         // ...
      }
   } else {
      if (kMaxTxnTimestamp == read_timestamp_) {
         // ...
      }
   }
}
```

Reply:
Also, I feel it would be nice to proactively make the same change in `GetEntityForUpdate` too

Reply:
Thanks a lot for this suggestion! I included this into a `SanityCheckReadTimestamp` method to be shared by them too.